### ISSUE TITLE
Vagrant cannot have a underscore in the environment name

### ISSUE THREAD

**Issue by alexmanly** [chef-cookbooks/delivery-cluster#170]
If you have an underscore in the environment name and you are using vagrant, then the builder node fails to provision.

**Comment by afiune** [comment]
This is very specific and it is not a restriction from `delivery-cluster` so I don't think this is a valid issue. I will mark it as that but let it be open for a while. 

Although thank you for opening issues as you encounter problems. I am thinking in creating a FAQ Markdown file for all this little/specific things. What do you think about that?

**Comment by alexmanly** [comment]
You're right this is a usability issue specifically when using rake to generate a vagrant environment.  However, it would be just as easy to put the validation in the drive_name case statement when 'vagrant' test for underscores in the environment name and exit with a msg if found.

**Comment by alexmanly** [comment]
Looking at other issues, looks like this is worth validating in code.  https://github.com/chef-cookbooks/delivery-cluster/issues/165
